src/org/apache/commons/httpclient/HttpMethodDirector.java

```diff
@@ -104,6 +108,8 @@
 
     private static final Log LOG = LogFactory.getLog(HttpMethodDirector.class);
 
+    private static List<HttpCookie> redirectCookies = new ArrayList<>();
```

Comment:
This will not work properly when multiple (independent) requests are sent at the same time.